pkgs/development/python-modules/poetry-dynamic-versioning/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, buildPythonPackage, fetchPypi
+, dunamai
+, jinja2
+, tomlkit
+}:
+
+buildPythonPackage rec {
+  pname = "poetry-dynamic-versioning";
+  version = "0.13.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5c0e7b22560db76812057ef95dadad662ecc63eb270145787eabe73da7c222f9";
+  };
+
+  propagatedBuildInputs = [
+    dunamai
+    jinja2
+    tomlkit
+  ];
+
```

Comment:
[Tests](https://github.com/mtkennerly/poetry-dynamic-versioning/tree/master/tests) would be available if the source from GitHub is used.